app/crud.py

```diff
@@ -23,7 +32,52 @@ def get_user_by_userid(db: Session, user_id: int):
 
 
 def login_user(db, username: str, password: str):
-    hashed_password = hashlib.sha256(password.encode('utf-8')).hexdigest()
-    return db.query(models.User).filter(
-        and_(models.User.password == hashed_password,
-             models.User.username == username)).first()
+    hashed_password = hashlib.sha256(password.encode("utf-8")).hexdigest()
+    return (
+        db.query(models.User)
+        .filter(
+            and_(
+                models.User.password == hashed_password,
+                models.User.username == username,
+            )
+        )
+        .first()
+    )
+
+
+def get_posts(db: Session, skip: int = 0, limit: int = 100):
+    return db.query(models.Post).offset(skip).limit(limit).all()
+
+
+def get_post_by_id(db: Session, post_id: int):
+    return db.query(models.Post).filter(models.Post.id == post_id).first()
+
+
+def create_post(db: Session, post: schema.PostCreate, user):
+    db_item = models.Post(**post.dict(), author_id=user.id)
+    db.add(db_item)
+    db.commit()
+    db.refresh(db_item)
+    return db_item
+
+
+def edit_post(db: Session, post: models.Post, post_data: schema.PostEdit):
+    if post_data.title is not None:
+        post.title = post_data.title
+    if post_data.type is not None:
+        post.type = post_data.type
```

Comment:
Should the type be changeable?

Reply:
Why not?

Reply:
I guess that's something to be addressed in the upcoming meeting. For now, it's ok.